include/openthread/platform/ble.h

```diff
@@ -0,0 +1,1039 @@
+
+    /**
+     * Unit used to calculate scan interval duration (0.625ms).
+     */
+    OT_BLE_SCAN_INTERVAL_UNIT = OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * Minimum allowed scan window in OT_BLE_TIMESLOT_UNIT units (2.5ms).
+     */
+    OT_BLE_SCAN_WINDOW_MIN = 0x0004,
+
+    /**
+     * Maximum allowed scan window in OT_BLE_TIMESLOT_UNIT units (10.24s).
+     */
+    OT_BLE_SCAN_WINDOW_MAX = 0x4000,
+
+    /**
+     * Unit used to calculate scan window duration (0.625ms).
+     */
+    OT_BLE_SCAN_WINDOW_UNIT = OT_BLE_TIMESLOT_UNIT,
+
+    /**
+     * BLE HCI code for remote user terminated connection.
+     */
+    OT_BLE_HCI_REMOTE_USER_TERMINATED = 0x13,
+
+    /**
+     * Value of invalid/unknown handle.
+     */
+    OT_BLE_INVALID_HANDLE = 0x0000,
+
+    /**
+     * Maximum size of BLE Characteristic.
+     */
+    OT_BLE_CHARACTERISTIC_MAX_LENGTH = 512,
+
+    /**
+     * Maximum value of ATT_MTU.
+     */
+    OT_BLE_ATT_MTU_MAX = 511,
+
+    /**
+     * Length of full BLE UUID in bytes.
+     */
+    OT_BLE_UUID_LENGTH = 16,
+
+    /**
+     * Uuid of Client Configuration Characteristic Descriptor.
+     */
+    OT_BLE_UUID_CCCD = 0x2902,
+
+};
+
+/**
+ * This enum represents BLE Device Address types.
+ *
+ */
+enum
+{
+    OT_BLE_ADDRESS_TYPE_PUBLIC                        = 0, ///< Bluetooth public device address.
+    OT_BLE_ADDRESS_TYPE_RANDOM_STATIC                 = 1, ///< Bluetooth random static address.
+    OT_BLE_ADDRESS_TYPE_RANDOM_PRIVATE_RESOLVABLE     = 2, ///< Bluetooth random private resolvable address.
+    OT_BLE_ADDRESS_TYPE_RANDOM_PRIVATE_NON_RESOLVABLE = 3, ///< Bluetooth random private non-resolvable address.
+};
+
+/**
+ * This enum represents characterstic properties.
+ *
+ * See v4.2 [Vol 3, Part G] 3.3.1.1 Characteristic Properties - Table 3.5
+ */
+enum
+{
+    /**
+     * If set, permits broadcasts of the Characteristic Value using Characteristic Configuration Descriptor.
+     */
+    OT_BLE_CHAR_PROP_BROADCAST = 0x01,
+
+    /**
+     * If set, permits reads of the Characteristic Value.
+     */
+    OT_BLE_CHAR_PROP_READ = 0x02,
+
+    /**
+     * If set, permit writes of the Characteristic Value without response.
+     */
+    OT_BLE_CHAR_PROP_WRITE_NO_RESPONSE = 0x04,
+
+    /**
+     * If set, permits writes of the Characteristic Value with response.
+     */
+    OT_BLE_CHAR_PROP_WRITE = 0x08,
+
+    /**
+     * If set, permits notifications of a Characteristic Value without acknowledgement.
+     */
+    OT_BLE_CHAR_PROP_NOTIFY = 0x10,
+
+    /**
+     * If set, permits indications of a Characteristic Value with acknowledgement.
+     */
+    OT_BLE_CHAR_PROP_INDICATE = 0x20,
+
+    /**
+     * If set, permits signed writes to the Characteristic Value.
+     */
+    OT_BLE_CHAR_PROP_AUTH_SIGNED_WRITE = 0x40,
+
+    /**
+     * If set, additional characteristic properties are defined in the Characteristic Extended Properties Descriptor.
+     */
+    OT_BLE_CHAR_PROP_EXTENDED = 0x80,
+
+};
```

Comment:
This enum looks like a set of bit masks but there is nothing in the description that indicates it as such.  If it is indeed a bitmask the format 1<<x is preferred over 0xYY. See types.h for examples.